### PR TITLE
fix: #3847 crash on session close

### LINKED ISSUE #3847: Crash on exit in [d82090658f] (from September 24): tr_session::tr_udp_core::~tr_udp_core()
This is yet a different line of code that is crashing.

- This radar #3847 is crashing on exit on `tr_session::tr_udp_core::~tr_udp_core() + 28 (tr-udp.cc:318)`
- #3825 was crashing on exit on `libtransmission::EvTimer::~EvTimer() + 14`
- #3811 was crashing in background on `tr_peerMgrGetDesiredAvailable(tr_torrent const*) + 560 (peer-mgr.cc:1656)`
- #3776 was crashing on exit on `tr_session::~tr_session() + 424`

```
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               Transmission [3911]
Path:                  /usr/local/Applications/Transmission.app/Contents/MacOS/Transmission
Identifier:            org.m0k.transmission
Version:               3.00+ (d82090658f)
Code Type:             ARM-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2022-09-27 23:18:03.3093 +0800
OS Version:            macOS 13.0 (22A5352e)
Report Version:        12
Anonymous UUID:        3AA30E44-F9A9-D0F9-693B-54C0766FE3B6

Sleep/Wake UUID:       55CF4238-3120-469E-97F9-3D5899621C8C

Time Awake Since Boot: 4900 seconds
Time Since Wake:       3755 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000

Application Specific Information:
abort() called


Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	       0x1929b3204 __pthread_kill + 8
1   libsystem_pthread.dylib       	       0x1929e9cec pthread_kill + 288
2   libsystem_c.dylib             	       0x1929232c8 abort + 180
3   libc++abi.dylib               	       0x1929a3b18 abort_message + 132
4   libc++abi.dylib               	       0x1929939f4 demangling_terminate_handler() + 312
5   libobjc.A.dylib               	       0x192697774 _objc_terminate() + 160
6   libc++abi.dylib               	       0x1929a2eb4 std::__terminate(void (*)()) + 20
7   libc++abi.dylib               	       0x1929a2e50 std::terminate() + 56
8   Transmission                  	       0x104be4460 tr_session::tr_udp_core::~tr_udp_core() + 28 (tr-udp.cc:318)
9   Transmission                  	       0x104b1d84c std::__1::default_delete<tr_session::tr_udp_core>::operator()(tr_session::tr_udp_core*) const + 44 (unique_ptr.h:57)
10  Transmission                  	       0x104af9cd4 std::__1::unique_ptr<tr_session::tr_udp_core, std::__1::default_delete<tr_session::tr_udp_core> >::reset(tr_session::tr_udp_core*) + 96 (unique_ptr.h:318)
11  Transmission                  	       0x104b17dd0 std::__1::unique_ptr<tr_session::tr_udp_core, std::__1::default_delete<tr_session::tr_udp_core> >::~unique_ptr() + 32 (unique_ptr.h:272)
12  Transmission                  	       0x104af52d8 std::__1::unique_ptr<tr_session::tr_udp_core, std::__1::default_delete<tr_session::tr_udp_core> >::~unique_ptr() + 28 (unique_ptr.h:272)
13  Transmission                  	       0x104b06c1c tr_session::~tr_session() + 404 (session.h:132)
14  Transmission                  	       0x104afaa4c tr_session::~tr_session() + 28 (session.h:132)
15  Transmission                  	       0x104afa844 tr_sessionClose(tr_session*) + 2712 (session.cc:1957)
16  Transmission                  	       0x1048253d4 -[Controller applicationWillTerminate:] + 2468 (Controller.mm:1044)
17  CoreFoundation                	       0x192abea1c __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 148
18  CoreFoundation                	       0x192b5c4a4 ___CFXRegistrationPost_block_invoke + 88
19  CoreFoundation                	       0x192b5c3ec _CFXRegistrationPost + 440
20  CoreFoundation                	       0x192a9003c _CFXNotificationPost + 708
21  Foundation                    	       0x19397ba3c -[NSNotificationCenter postNotificationName:object:userInfo:] + 88
22  AppKit                        	       0x195f30cec -[NSApplication terminate:] + 1092
23  AppKit                        	       0x196055304 -[NSApplication _terminateFromSender:askIfShouldTerminate:saveWindows:] + 124
24  AppKit                        	       0x1960551fc __52-[NSApplication(NSAppleEventHandling) _handleAEQuit]_block_invoke + 56
25  AppKit                        	       0x1960a9734 ___NSMainRunLoopPerformBlockInModes_block_invoke + 44
26  CoreFoundation                	       0x192ac9a5c __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 28
27  CoreFoundation                	       0x192ac9970 __CFRunLoopDoBlocks + 368
28  CoreFoundation                	       0x192ac87bc __CFRunLoopRun + 820
29  CoreFoundation                	       0x192ac7d34 CFRunLoopRunSpecific + 612
30  HIToolbox                     	       0x19c1ea2d4 RunCurrentEventLoopInMode + 292
31  HIToolbox                     	       0x19c1e9f64 ReceiveNextEventCommon + 236
32  HIToolbox                     	       0x19c1e9e60 _BlockUntilNextEventMatchingListInModeWithFilter + 72
33  AppKit                        	       0x195d1b7e4 _DPSNextEvent + 632
34  AppKit                        	       0x195d1a974 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 728
35  AppKit                        	       0x195d0eda4 -[NSApplication run] + 464
36  AppKit                        	       0x195ce6314 NSApplicationMain + 880
37  Transmission                  	       0x1048b814c main + 36 (main.mm:9)
38  dyld                          	       0x1926bfe50 start + 2544

Thread 1:
0   libsystem_pthread.dylib       	       0x1929e4e18 start_wqthread + 0

Thread 2:: com.apple.NSEventThread
0   libsystem_kernel.dylib        	       0x1929aadd0 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x1929bc8b8 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x1929b3618 mach_msg_overwrite + 540
3   libsystem_kernel.dylib        	       0x1929ab14c mach_msg + 24
4   CoreFoundation                	       0x192aca06c __CFRunLoopServiceMachPort + 160
5   CoreFoundation                	       0x192ac8958 __CFRunLoopRun + 1232
6   CoreFoundation                	       0x192ac7d34 CFRunLoopRunSpecific + 612
7   AppKit                        	       0x195e46070 _NSEventThread + 172
8   libsystem_pthread.dylib       	       0x1929ea06c _pthread_start + 148
9   libsystem_pthread.dylib       	       0x1929e4e2c thread_start + 8

Thread 3:
0   libsystem_pthread.dylib       	       0x1929e4e18 start_wqthread + 0


Thread 0 crashed with ARM Thread State (64-bit):
    x0: 0x0000000000000000   x1: 0x0000000000000000   x2: 0x0000000000000000   x3: 0x0000000000000000
    x4: 0x00000001929a70c8   x5: 0x000000016b5f83c0   x6: 0x0000000000000074   x7: 0x000000016b5f86f8
    x8: 0x69d7ed56138cab35   x9: 0x69d7ed57fd98bdb5  x10: 0x0000000000000200  x11: 0x000000000000002d
   x12: 0x000000000000002d  x13: 0x00000000001ff800  x14: 0x00000000000007fb  x15: 0x00000000f2e1a055
   x16: 0x0000000000000148  x17: 0x00000001f2f3f610  x18: 0x0000000000000000  x19: 0x0000000000000006
   x20: 0x00000001ee141680  x21: 0x0000000000000103  x22: 0x00000001ee141760  x23: 0x00006000038309e0
   x24: 0x0000000000000000  x25: 0x0000000000041400  x26: 0x0000000000000000  x27: 0x0000004400000045
   x28: 0x000000016b5f9588   fp: 0x000000016b5f8330   lr: 0x00000001929e9cec
    sp: 0x000000016b5f8310   pc: 0x00000001929b3204 cpsr: 0x40001000
   far: 0x0000000104daaf0c  esr: 0x56000080  Address size fault

Binary Images:
       0x1929aa000 -        0x1929e2feb libsystem_kernel.dylib (*) <b54eb397-ff4e-3415-8536-d98a24ab7d35> /usr/lib/system/libsystem_kernel.dylib
       0x1929e3000 -        0x1929efffb libsystem_pthread.dylib (*) <93540939-bf1b-3a4a-bb32-923eb7326b45> /usr/lib/system/libsystem_pthread.dylib
       0x1928aa000 -        0x19292aff3 libsystem_c.dylib (*) <9da4fe76-ad8a-3b91-b67a-5190259fad11> /usr/lib/system/libsystem_c.dylib
       0x192992000 -        0x1929a9ffb libc++abi.dylib (*) <740e4317-0cc2-3122-bafa-9ed245e16608> /usr/lib/libc++abi.dylib
       0x192674000 -        0x1926b9d9f libobjc.A.dylib (*) <0aaada28-b470-3943-87d0-344cc9d765a4> /usr/lib/libobjc.A.dylib
       0x104804000 -        0x104daffff org.m0k.transmission (3.00+) <704410f7-a434-3ae4-b3e3-472342f91c33> /usr/local/Applications/Transmission.app/Contents/MacOS/Transmission
       0x192a48000 -        0x192f1ffff com.apple.CoreFoundation (6.9) <eef8b379-1217-304b-b53d-70cd1e1f32c6> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
       0x193972000 -        0x1943aafff com.apple.Foundation (6.9) <13f36666-dc15-3bdf-8c77-3cdc37d43315> /System/Library/Frameworks/Foundation.framework/Versions/C/Foundation
       0x195ce2000 -        0x196be5fff com.apple.AppKit (6.9) <295edd66-3303-3ff6-8eb1-34912866ae72> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
       0x19c1b8000 -        0x19c4ebfff com.apple.HIToolbox (2.1.1) <7d970c28-21b2-3e40-a008-b67c62fd083f> /System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/HIToolbox.framework/Versions/A/HIToolbox
       0x1926ba000 -        0x1927442a3 dyld (*) <a806cc48-a704-37f7-923b-8158a3d68c05> /usr/lib/dyld

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=1.3G resident=0K(0%) swapped_out_or_unallocated=1.3G(100%)
Writable regions: Total=1.6G written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=1.6G(100%)

                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Accelerate framework               768K        6 
Activity Tracing                   256K        1 
CG image                           240K        8 
ColorSync                          544K       26 
CoreAnimation                     5248K       40 
CoreGraphics                        32K        2 
CoreUI image data                  752K       11 
Foundation                          16K        1 
Image IO                            80K        5 
Kernel Alloc Once                   32K        1 
MALLOC                           280.3M       49 
MALLOC guard page                  192K       10 
MALLOC_MEDIUM (reserved)         960.0M        8         reserved VM address space (unallocated)
MALLOC_NANO (reserved)           384.0M        1         reserved VM address space (unallocated)
STACK GUARD                       56.1M        4 
Stack                             9808K        4 
VM_ALLOCATE                        224K        7 
VM_ALLOCATE (reserved)              64K        1         reserved VM address space (unallocated)
__AUTH                            1730K      322 
__AUTH_CONST                      23.5M      535 
__CTF                               756        1 
__DATA                            11.6M      522 
__DATA_CONST                      26.4M      539 
__DATA_DIRTY                      1913K      212 
__FONT_DATA                        2352        1 
__LINKEDIT                       773.7M        4 
__OBJC_CONST                      4322K      288 
__OBJC_RO                         65.0M        1 
__OBJC_RW                         1977K        1 
__TEXT                           566.3M      561 
dyld private memory                256K        1 
mapped file                      187.6M       54 
shared memory                      912K       16 
===========                     =======  ======= 
TOTAL                              3.3G     3243 
TOTAL, minus reserved VM space     2.0G     3243 



-----------
Full Report
-----------

{"app_name":"Transmission","timestamp":"2022-09-27 23:18:14.00 +0800","app_version":"3.00+","slice_uuid":"704410f7-a434-3ae4-b3e3-472342f91c33","build_version":"d82090658f","platform":1,"bundleID":"org.m0k.transmission","share_with_app_devs":1,"is_first_party":0,"bug_type":"309","os_version":"macOS 13.0 (22A5352e)","roots_installed":0,"name":"Transmission","incident_id":"8324D9AA-FFF4-4E9E-A596-075B06F3F829"}
{
  "uptime" : 4900,
  "procRole" : "Background",
  "version" : 2,
  "userID" : 501,
  "deployVersion" : 210,
  "modelCode" : "Macmini9,1",
  "coalitionID" : 644,
  "osVersion" : {
    "train" : "macOS 13.0",
    "build" : "22A5352e",
    "releaseType" : "User"
  },
  "captureTime" : "2022-09-27 23:18:03.3093 +0800",
  "incident" : "8324D9AA-FFF4-4E9E-A596-075B06F3F829",
  "pid" : 3911,
  "translated" : false,
  "cpuType" : "ARM-64",
  "roots_installed" : 0,
  "bug_type" : "309",
  "procLaunch" : "2022-09-27 23:17:22.9046 +0800",
  "procStartAbsTime" : 118661295531,
  "procExitAbsTime" : 119630937862,
  "procName" : "Transmission",
  "procPath" : "\/usr\/local\/Applications\/Transmission.app\/Contents\/MacOS\/Transmission",
  "bundleInfo" : {"CFBundleShortVersionString":"3.00+","CFBundleVersion":"d82090658f","CFBundleIdentifier":"org.m0k.transmission"},
  "storeInfo" : {"deviceIdentifierForVendor":"7E344BCC-1489-5E9A-800F-299F8CB9625D","thirdParty":true},
  "parentProc" : "launchd",
  "parentPid" : 1,
  "coalitionName" : "org.m0k.transmission",
  "crashReporterKey" : "3AA30E44-F9A9-D0F9-693B-54C0766FE3B6",
  "wakeTime" : 3755,
  "sleepWakeUUID" : "55CF4238-3120-469E-97F9-3D5899621C8C",
  "sip" : "enabled",
  "exception" : {"codes":"0x0000000000000000, 0x0000000000000000","rawCodes":[0,0],"type":"EXC_CRASH","signal":"SIGABRT"},
  "asi" : {"libsystem_c.dylib":["abort() called"]},
  "extMods" : {"caller":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"system":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"targeted":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"warnings":0},
  "faultingThread" : 0,
  "threads" : [{"triggered":true,"id":80199,"threadState":{"x":[{"value":0},{"value":0},{"value":0},{"value":0},{"value":6754562248},{"value":6096389056},{"value":116},{"value":6096389880},{"value":7626825447926180661},{"value":7626825456147807669},{"value":512},{"value":45},{"value":45},{"value":2095104},{"value":2043},{"value":4074872917},{"value":328},{"value":8371041808},{"value":0},{"value":6},{"value":8289261184,"symbolLocation":0,"symbol":"_main_thread"},{"value":259},{"value":8289261408,"symbolLocation":224,"symbol":"_main_thread"},{"value":105553175185888},{"value":0},{"value":267264},{"value":0},{"value":292057776197},{"value":6096393608}],"flavor":"ARM_THREAD_STATE64","lr":{"value":6754835692},"cpsr":{"value":1073745920},"fp":{"value":6096388912},"sp":{"value":6096388880},"esr":{"value":1442840704,"description":" Address size fault"},"pc":{"value":6754611716,"matchesCrashFrame":1},"far":{"value":4376407820}},"queue":"com.apple.main-thread","frames":[{"imageOffset":37380,"symbol":"__pthread_kill","symbolLocation":8,"imageIndex":0},{"imageOffset":27884,"symbol":"pthread_kill","symbolLocation":288,"imageIndex":1},{"imageOffset":496328,"symbol":"abort","symbolLocation":180,"imageIndex":2},{"imageOffset":72472,"symbol":"abort_message","symbolLocation":132,"imageIndex":3},{"imageOffset":6644,"symbol":"demangling_terminate_handler()","symbolLocation":312,"imageIndex":3},{"imageOffset":145268,"symbol":"_objc_terminate()","symbolLocation":160,"imageIndex":4},{"imageOffset":69300,"symbol":"std::__terminate(void (*)())","symbolLocation":20,"imageIndex":3},{"imageOffset":69200,"symbol":"std::terminate()","symbolLocation":56,"imageIndex":3},{"imageOffset":4064352,"sourceLine":318,"sourceFile":"tr-udp.cc","symbol":"tr_session::tr_udp_core::~tr_udp_core()","imageIndex":5,"symbolLocation":28},{"imageOffset":3250252,"sourceLine":57,"sourceFile":"unique_ptr.h","symbol":"std::__1::default_delete<tr_session::tr_udp_core>::operator()(tr_session::tr_udp_core*) const","imageIndex":5,"symbolLocation":44},{"imageOffset":3103956,"sourceLine":318,"sourceFile":"unique_ptr.h","symbol":"std::__1::unique_ptr<tr_session::tr_udp_core, std::__1::default_delete<tr_session::tr_udp_core> >::reset(tr_session::tr_udp_core*)","imageIndex":5,"symbolLocation":96},{"imageOffset":3227088,"sourceLine":272,"sourceFile":"unique_ptr.h","symbol":"std::__1::unique_ptr<tr_session::tr_udp_core, std::__1::default_delete<tr_session::tr_udp_core> >::~unique_ptr()","imageIndex":5,"symbolLocation":32},{"imageOffset":3085016,"sourceLine":272,"sourceFile":"unique_ptr.h","symbol":"std::__1::unique_ptr<tr_session::tr_udp_core, std::__1::default_delete<tr_session::tr_udp_core> >::~unique_ptr()","imageIndex":5,"symbolLocation":28},{"imageOffset":3157020,"sourceLine":132,"sourceFile":"session.h","symbol":"tr_session::~tr_session()","imageIndex":5,"symbolLocation":404},{"imageOffset":3107404,"sourceLine":132,"sourceFile":"session.h","symbol":"tr_session::~tr_session()","imageIndex":5,"symbolLocation":28},{"imageOffset":3106884,"sourceLine":1957,"sourceFile":"session.cc","symbol":"tr_sessionClose(tr_session*)","imageIndex":5,"symbolLocation":2712},{"imageOffset":136148,"sourceLine":1044,"sourceFile":"Controller.mm","symbol":"-[Controller applicationWillTerminate:]","imageIndex":5,"symbolLocation":2468},{"imageOffset":485916,"symbol":"__CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__","symbolLocation":148,"imageIndex":6},{"imageOffset":1131684,"symbol":"___CFXRegistrationPost_block_invoke","symbolLocation":88,"imageIndex":6},{"imageOffset":1131500,"symbol":"_CFXRegistrationPost","symbolLocation":440,"imageIndex":6},{"imageOffset":294972,"symbol":"_CFXNotificationPost","symbolLocation":708,"imageIndex":6},{"imageOffset":39484,"symbol":"-[NSNotificationCenter postNotificationName:object:userInfo:]","symbolLocation":88,"imageIndex":7},{"imageOffset":2419948,"symbol":"-[NSApplication terminate:]","symbolLocation":1092,"imageIndex":8},{"imageOffset":3617540,"symbol":"-[NSApplication _terminateFromSender:askIfShouldTerminate:saveWindows:]","symbolLocation":124,"imageIndex":8},{"imageOffset":3617276,"symbol":"__52-[NSApplication(NSAppleEventHandling) _handleAEQuit]_block_invoke","symbolLocation":56,"imageIndex":8},{"imageOffset":3962676,"symbol":"___NSMainRunLoopPerformBlockInModes_block_invoke","symbolLocation":44,"imageIndex":8},{"imageOffset":531036,"symbol":"__CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__","symbolLocation":28,"imageIndex":6},{"imageOffset":530800,"symbol":"__CFRunLoopDoBlocks","symbolLocation":368,"imageIndex":6},{"imageOffset":526268,"symbol":"__CFRunLoopRun","symbolLocation":820,"imageIndex":6},{"imageOffset":523572,"symbol":"CFRunLoopRunSpecific","symbolLocation":612,"imageIndex":6},{"imageOffset":205524,"symbol":"RunCurrentEventLoopInMode","symbolLocation":292,"imageIndex":9},{"imageOffset":204644,"symbol":"ReceiveNextEventCommon","symbolLocation":236,"imageIndex":9},{"imageOffset":204384,"symbol":"_BlockUntilNextEventMatchingListInModeWithFilter","symbolLocation":72,"imageIndex":9},{"imageOffset":235492,"symbol":"_DPSNextEvent","symbolLocation":632,"imageIndex":8},{"imageOffset":231796,"symbol":"-[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:]","symbolLocation":728,"imageIndex":8},{"imageOffset":183716,"symbol":"-[NSApplication run]","symbolLocation":464,"imageIndex":8},{"imageOffset":17172,"symbol":"NSApplicationMain","symbolLocation":880,"imageIndex":8},{"imageOffset":737612,"sourceLine":9,"sourceFile":"main.mm","symbol":"main","imageIndex":5,"symbolLocation":36},{"imageOffset":24144,"symbol":"start","symbolLocation":2544,"imageIndex":10}]},{"id":82808,"frames":[{"imageOffset":7704,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":1}]},{"id":82851,"name":"com.apple.NSEventThread","frames":[{"imageOffset":3536,"symbol":"mach_msg2_trap","symbolLocation":8,"imageIndex":0},{"imageOffset":75960,"symbol":"mach_msg2_internal","symbolLocation":80,"imageIndex":0},{"imageOffset":38424,"symbol":"mach_msg_overwrite","symbolLocation":540,"imageIndex":0},{"imageOffset":4428,"symbol":"mach_msg","symbolLocation":24,"imageIndex":0},{"imageOffset":532588,"symbol":"__CFRunLoopServiceMachPort","symbolLocation":160,"imageIndex":6},{"imageOffset":526680,"symbol":"__CFRunLoopRun","symbolLocation":1232,"imageIndex":6},{"imageOffset":523572,"symbol":"CFRunLoopRunSpecific","symbolLocation":612,"imageIndex":6},{"imageOffset":1458288,"symbol":"_NSEventThread","symbolLocation":172,"imageIndex":8},{"imageOffset":28780,"symbol":"_pthread_start","symbolLocation":148,"imageIndex":1},{"imageOffset":7724,"symbol":"thread_start","symbolLocation":8,"imageIndex":1}]},{"id":83773,"frames":[{"imageOffset":7704,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":1}]}],
  "usedImages" : [
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6754574336,
    "size" : 233452,
    "uuid" : "b54eb397-ff4e-3415-8536-d98a24ab7d35",
    "path" : "\/usr\/lib\/system\/libsystem_kernel.dylib",
    "name" : "libsystem_kernel.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6754807808,
    "size" : 53244,
    "uuid" : "93540939-bf1b-3a4a-bb32-923eb7326b45",
    "path" : "\/usr\/lib\/system\/libsystem_pthread.dylib",
    "name" : "libsystem_pthread.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6753525760,
    "size" : 528372,
    "uuid" : "9da4fe76-ad8a-3b91-b67a-5190259fad11",
    "path" : "\/usr\/lib\/system\/libsystem_c.dylib",
    "name" : "libsystem_c.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6754476032,
    "size" : 98300,
    "uuid" : "740e4317-0cc2-3122-bafa-9ed245e16608",
    "path" : "\/usr\/lib\/libc++abi.dylib",
    "name" : "libc++abi.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6751207424,
    "size" : 286112,
    "uuid" : "0aaada28-b470-3943-87d0-344cc9d765a4",
    "path" : "\/usr\/lib\/libobjc.A.dylib",
    "name" : "libobjc.A.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4370481152,
    "CFBundleShortVersionString" : "3.00+",
    "CFBundleIdentifier" : "org.m0k.transmission",
    "size" : 5947392,
    "uuid" : "704410f7-a434-3ae4-b3e3-472342f91c33",
    "path" : "\/usr\/local\/Applications\/Transmission.app\/Contents\/MacOS\/Transmission",
    "name" : "Transmission",
    "CFBundleVersion" : "d82090658f"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6755221504,
    "CFBundleShortVersionString" : "6.9",
    "CFBundleIdentifier" : "com.apple.CoreFoundation",
    "size" : 5079040,
    "uuid" : "eef8b379-1217-304b-b53d-70cd1e1f32c6",
    "path" : "\/System\/Library\/Frameworks\/CoreFoundation.framework\/Versions\/A\/CoreFoundation",
    "name" : "CoreFoundation",
    "CFBundleVersion" : "1951"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6771122176,
    "CFBundleShortVersionString" : "6.9",
    "CFBundleIdentifier" : "com.apple.Foundation",
    "size" : 10719232,
    "uuid" : "13f36666-dc15-3bdf-8c77-3cdc37d43315",
    "path" : "\/System\/Library\/Frameworks\/Foundation.framework\/Versions\/C\/Foundation",
    "name" : "Foundation",
    "CFBundleVersion" : "1951"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6808281088,
    "CFBundleShortVersionString" : "6.9",
    "CFBundleIdentifier" : "com.apple.AppKit",
    "size" : 15745024,
    "uuid" : "295edd66-3303-3ff6-8eb1-34912866ae72",
    "path" : "\/System\/Library\/Frameworks\/AppKit.framework\/Versions\/C\/AppKit",
    "name" : "AppKit",
    "CFBundleVersion" : "2285.10"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6914015232,
    "CFBundleShortVersionString" : "2.1.1",
    "CFBundleIdentifier" : "com.apple.HIToolbox",
    "size" : 3358720,
    "uuid" : "7d970c28-21b2-3e40-a008-b67c62fd083f",
    "path" : "\/System\/Library\/Frameworks\/Carbon.framework\/Versions\/A\/Frameworks\/HIToolbox.framework\/Versions\/A\/HIToolbox",
    "name" : "HIToolbox"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 6751494144,
    "size" : 565924,
    "uuid" : "a806cc48-a704-37f7-923b-8158a3d68c05",
    "path" : "\/usr\/lib\/dyld",
    "name" : "dyld"
  }
],
  "sharedCache" : {
  "base" : 6750846976,
  "size" : 3408183296,
  "uuid" : "4d3af3a2-3e53-371c-94f3-ac97bb52ebc8"
},
  "vmSummary" : "ReadOnly portion of Libraries: Total=1.3G resident=0K(0%) swapped_out_or_unallocated=1.3G(100%)\nWritable regions: Total=1.6G written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=1.6G(100%)\n\n                                VIRTUAL   REGION \nREGION TYPE                        SIZE    COUNT (non-coalesced) \n===========                     =======  ======= \nAccelerate framework               768K        6 \nActivity Tracing                   256K        1 \nCG image                           240K        8 \nColorSync                          544K       26 \nCoreAnimation                     5248K       40 \nCoreGraphics                        32K        2 \nCoreUI image data                  752K       11 \nFoundation                          16K        1 \nImage IO                            80K        5 \nKernel Alloc Once                   32K        1 \nMALLOC                           280.3M       49 \nMALLOC guard page                  192K       10 \nMALLOC_MEDIUM (reserved)         960.0M        8         reserved VM address space (unallocated)\nMALLOC_NANO (reserved)           384.0M        1         reserved VM address space (unallocated)\nSTACK GUARD                       56.1M        4 \nStack                             9808K        4 \nVM_ALLOCATE                        224K        7 \nVM_ALLOCATE (reserved)              64K        1         reserved VM address space (unallocated)\n__AUTH                            1730K      322 \n__AUTH_CONST                      23.5M      535 \n__CTF                               756        1 \n__DATA                            11.6M      522 \n__DATA_CONST                      26.4M      539 \n__DATA_DIRTY                      1913K      212 \n__FONT_DATA                        2352        1 \n__LINKEDIT                       773.7M        4 \n__OBJC_CONST                      4322K      288 \n__OBJC_RO                         65.0M        1 \n__OBJC_RW                         1977K        1 \n__TEXT                           566.3M      561 \ndyld private memory                256K        1 \nmapped file                      187.6M       54 \nshared memory                      912K       16 \n===========                     =======  ======= \nTOTAL                              3.3G     3243 \nTOTAL, minus reserved VM space     2.0G     3243 \n",
  "legacyInfo" : {
  "threadTriggered" : {
    "queue" : "com.apple.main-thread"
  }
},
  "trialInfo" : {
  "rollouts" : [
    {
      "rolloutId" : "610d52e1fc54bc3389840408",
      "factorPackIds" : {
        "SIRI_UNDERSTANDING_ASR_ASSISTANT" : "6306a428b4688478d744c68e",
        "SIRI_UNDERSTANDING_MORPHUN" : "62ec7220c682040ba94e6a20"
      },
      "deploymentId" : 250000403
    },
    {
      "rolloutId" : "60da5e84ab0ca017dace9abf",
      "factorPackIds" : {

      },
      "deploymentId" : 250000007
    }
  ],
  "experiments" : [

  ]
}
}

Model: Macmini9,1, BootROM 8419.40.95.0.1, proc 8:4:4 processors, 16 GB, SMC 
Graphics: Apple M1, Apple M1, Built-In
Display: Mi TV, 1920 x 1080 (1080p FHD - Full High Definition), Main, MirrorOff, Online
Memory Module: LPDDR4, Hynix
AirPort: spairport_wireless_card_type_wifi (0x14E4, 0x4378), wl0: Jul 27 2022 06:39:54 version 18.20.377.0.7.8.142 FWID 01-12d8011f
Bluetooth: Version (null), 0 services, 0 devices, 0 incoming serial ports
Network Service: Wi-Fi, AirPort, en1
USB Device: USB31Bus
USB Device: Portable
USB Device: USB31Bus
USB Device: Magic Keyboard with Numeric Keypad
USB Device: USB30Bus
USB Device: BUP Portable
USB Device: One Touch w/PW
Thunderbolt Bus: Mac mini, Apple Inc.
Thunderbolt Bus: Mac mini, Apple Inc.
```

### PR DESCRIPTION
**Pull request by ckerr**
Fixes #3847.

There is still more work to be done on the clearinghouse of issues that @Coeur listed in 3847 but here's a down payment :smile_cat:

### CODE REVIEW

**Comment by Harm133** [comment]
Can you take another look at this? Getting a SIGSEGV since this change:

```
Thread 7 "transmission-qt" received signal SIGSEGV, Segmentation fault.
tr_dhtCallback (session=session@entry=0x555555e75e30, buf=buf@entry=0x7fffe97fc880 "**Edited Out**", buflen=<optimized out>, from=from@entry=0x7fffe97fc800, fromlen=<optimized out>) at /home/harm/Source/transmission-self/src/libtransmission/tr-dht.cc:793
793     dht_timer->startSingleShot(std::chrono::duration_cast<std::chrono::milliseconds>(interval));
(gdb) bt
#0  tr_dhtCallback(tr_session*, unsigned char*, int, sockaddr*, unsigned int)
    (session=session@entry=0x555555e75e30, buf=buf@entry=0x7fffe97fc880 "**Edited Out**", buflen=<optimized out>, from=from@entry=0x7fffe97fc800, fromlen=<optimized out>)
    at /home/harm/Source/transmission-self/src/libtransmission/tr-dht.cc:793
#1  0x00005555556cf527 in event_callback(int, short, void*) (s=<optimized out>, type=<optimized out>, vsession=0x555555e75e30) at /usr/include/c++/12.2.0/array:283
#2  0x00007ffff7ee5c45 in  () at /usr/lib/libevent-2.1.so.7
#3  0x00007ffff7ee73af in event_base_loop () at /usr/lib/libevent-2.1.so.7
#4  0x00005555556d3b3c in libeventThreadFunc(tr_event_handle*) (events=0x555555e77460) at /home/harm/Source/transmission-self/src/libtransmission/trevent.cc:202
#5  0x00007ffff60d62f3 in std::execute_native_thread_routine(void*) (__p=0x555555e73a50) at /usr/src/debug/gcc/libstdc++-v3/src/c++11/thread.cc:82
#6  0x00007ffff5e9f74d in  () at /usr/lib/libc.so.6
#7  0x00007ffff5f21700 in  () at /usr/lib/libc.so.6
```

**Comment by GaryElshaw** [comment]
Same. macOS 12.6 

[transmission-crash-on-launch.txt](https://github.com/transmission/transmission/files/9679669/transmission-crash-on-launch.txt)

File: libtransmission/tr-dht.cc [hunk 1]
```diff
@@ -216,18 +381,20 @@ static void dht_bootstrap(tr_session* session, std::vector<uint8_t> nodes, std::
             nap(15);
         }
 
-        if (bootstrap_done(my_session, 0))
+        if (isBootstrapDone(session, 0))
         {
             break;
         }
     }
+    TR_ASSERT(walk4 = std::data(nodes4) + std::size(nodes4));
```
Comment (nevack) [inline libtransmission/tr-dht.cc:389]: @ckerr Is it supposed to be an equality check, not assignment?
It crashes for me when empty `nodes4`/`nodes6` are passed.
`std::data` returns `nullptr`, this whole expr is evaluated to `nullptr + 0` -> `nullptr` -> `0` -> `false` and `TR_ASSERT` fails.

Comment (ckerr) [reply libtransmission/tr-dht.cc:389]: Fixed by @Coeur in f27e1b52f46c0ea207a195af6ee1940bc1bd8ed8

**Pull request merged**